CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+    # doesn't currently compile ? :
+    $<$<BOOL:${verify_nodeobject_keys}>:RIPPLE_VERIFY_NODEOBJECT_KEYS=1>)
+target_compile_options (opts
+  INTERFACE
+    $<$<AND:$<BOOL:${is_gcc}>,$<COMPILE_LANGUAGE:CXX>>:-Wsuggest-override>
+    $<$<BOOL:${perf}>:-fno-omit-frame-pointer>
+    $<$<BOOL:${coverage}>:-fprofile-arcs -ftest-coverage>
+    $<$<BOOL:${profile}>:-p -pg>)
+
+target_link_libraries (opts
+  INTERFACE
+    $<$<BOOL:${coverage}>:-fprofile-arcs -ftest-coverage>
+    $<$<BOOL:${profile}>:-p -pg>)
+
+if (jemalloc)
+  find_package (jemalloc REQUIRED)
+  target_compile_definitions (opts INTERFACE PROFILE_JEMALLOC)
+  target_include_directories (opts SYSTEM INTERFACE ${JEMALLOC_INCLUDE_DIRS})
+  target_link_libraries (opts INTERFACE ${JEMALLOC_LIBRARIES})
+  get_filename_component (JEMALLOC_LIB_PATH ${JEMALLOC_LIBRARIES} DIRECTORY)
+  ## TODO see if we can use the BUILD_RPATH target property (is it transitive?)
+  set (CMAKE_BUILD_RPATH ${CMAKE_BUILD_RPATH} ${JEMALLOC_LIB_PATH})
+endif ()
+
+if (san)
+  target_compile_options (opts
+    INTERFACE
+      -fsanitize=${san}
+      -fno-omit-frame-pointer)
+  target_compile_definitions (opts
+    INTERFACE
+      $<$<STREQUAL:${san},address>:SANITIZER=ASAN>
+      $<$<STREQUAL:${san},thread>:SANITIZER=TSAN>
+      $<$<STREQUAL:${san},memory>:SANITIZER=MSAN>
+      $<$<STREQUAL:${san},undefined>:SANITIZER=UBSAN>)
+  target_link_libraries (opts
+    INTERFACE
+      -fsanitize=${san}
+      $<$<AND:$<BOOL:${is_gcc}>,$<STREQUAL:${san},address>>:asan>
+      $<$<AND:$<BOOL:${is_gcc}>,$<STREQUAL:${san},thread>>:tsan>
+      $<$<AND:$<BOOL:${is_gcc}>,$<STREQUAL:${san},memory>>:msan>
+      $<$<AND:$<BOOL:${is_gcc}>,$<STREQUAL:${san},undefined>>:ubsan>)
+endif ()
+
+#[===================================================================[
+   rippled transitive library deps via an interface library
+#]===================================================================]
+
+add_library (ripple_syslibs INTERFACE)
+add_library (Ripple::syslibs ALIAS ripple_syslibs)
+target_link_libraries (ripple_syslibs
+  INTERFACE
+    $<$<BOOL:${MSVC}>:
+      legacy_stdio_definitions.lib
+      Shlwapi
+      kernel32
+      user32
+      gdi32
+      winspool
+      comdlg32
+      advapi32
+      shell32
+      ole32
+      oleaut32
+      uuid
+      odbc32
+      odbccp32
+      crypt32
+    >
+    $<$<NOT:$<BOOL:${MSVC}>>:dl>
+    $<$<NOT:$<OR:$<BOOL:${MSVC}>,$<BOOL:${APPLE}>>>:rt>)
+
+if (NOT MSVC)
+  set (THREADS_PREFER_PTHREAD_FLAG ON)
+  find_package (Threads)
+  target_link_libraries (ripple_syslibs INTERFACE Threads::Threads)
+endif ()
+
+add_library (ripple_libs INTERFACE)
+add_library (Ripple::libs ALIAS ripple_libs)
+target_link_libraries (ripple_libs INTERFACE Ripple::syslibs)
+
+#[===================================================================[
+   NIH dep: boost
+#]===================================================================]
+
+if ((NOT DEFINED BOOST_ROOT) AND (DEFINED ENV{BOOST_ROOT}))
```

Comment:
Can you help me understand why we support both CMake defines *and* environment variables? I've been wondering it since I first jumped into the codebase. It seems like we could just get away with CMake defines at `configure` time.

Reply:
yes, it's legacy...but I think some devs find it easier to set this ENV whenever they upgrade boost and not have to pass it each time they configure.